src/graphql.tsx

```diff
@@ -635,7 +635,10 @@ class ObservableQueryRecycler {
   public recycle (observableQuery: ObservableQuery<any>): void {
     // Stop the query from polling when we recycle. Polling may resume when we
     // reuse it and call `setOptions`.
-    observableQuery.stopPolling();
+    observableQuery.setOptions({
+      fetchPolicy: 'cache-only',
```

Comment:
The issue with merging options is that you cannot _remove_ any - only update and add. I think `fetchPolicy` is omitted by default. Maybe it's enough to set it to `undefined` instead? But then it's a less straight forward procedure to store the old options and later restore them if the values needs to be reasoned about.

But if it's a sound approach to simply override the `observableQ.options` before calling `setOptions` (like I do now, that could be done with the snapshotted options too of course).

I haven't studied the whole thing to fully reason about the recycler. However, the current issue with `fetchPolicy` and refetching on`resetStore` is giving us real trouble. Given the issue at hand the limitations of `apollo-client`'s `setOptions` are my only concern.